show/main.py

```diff
@@ -129,6 +170,112 @@ def run_command(command, display_cmd=False):
     if rc != 0:
         sys.exit(rc)
 
+
+def get_interface_mode():
+    mode = os.getenv('SONIC_CLI_IFACE_MODE')
+    if mode is None:
+        mode = "default"
+    return mode
+
+
+# Global Alias object
+_alias = Alias()
+
+
+def run_command_in_alias_mode(command, command_type=0, linux_command=False):
+    """Run command and replace all instances of SONiC interface names
+       in output with vendor-sepecific interface aliases.
+    """
+
+    process = subprocess.Popen(command, shell=True, stdout=subprocess.PIPE)
+
+    while True:
+        output = process.stdout.readline()
+        if output == '' and process.poll() is not None:
+            break
+        if output:
+            if linux_command:
+                for port_name in _alias.port_dict.keys():
+                    regex = re.compile(r"\b{}\b".format(port_name))
+                    result = re.findall(regex, output)
+                    if result:
+                        interface_name = ''.join(result)
+                        output = output.replace(
+                            interface_name, _alias.interface_name_to_alias(
+                                interface_name))
+                click.echo(output.rstrip('\n'))
+
+            else:
+                output = output.lstrip()
+                index = 0
+
+                if command_type == 1:
```

Comment:
`command_type` values as simple integers are not clearly defined (what does "1" mean? etc.). This makes them confusing. Also, if we have one "command_type" for each command, this will become unwieldy. It would be better if you could key off the command string itself, and not rely on an integer which represents the command.

Reply:
This is good one. Initially I thougt of using enums,  but due to lack of package I ruled out the option. As suggested I used the command strings directly in the function and included in the latest commit.